src/com/maddyhome/idea/vim/group/EditorGroup.java

```diff
@@ -206,10 +209,6 @@ public void closeEditorSearchSession(@NotNull Editor editor) {
     }
   }
 
-  public boolean isBarCursorSettings() {
-    return !EditorSettingsExternalizable.getInstance().isBlockCursor();
-  }
-
   public void editorCreated(@NotNull Editor editor) {
```

Comment:
No. For 212, this setting is mostly ignored. It controls whether the default caret is block or bar, and we don't use the default caret. If the user changes this setting during a session, the editor will redraw and use the default caret. This will remain visible until the user causes the Vim caret to change - changing mode or mouse drag.

Reply:
Now, wait, I believe we use this property. In this way, the user can keep the block caret in insert mode. I don't remember how this support was enabled, was it a user request or something.

Reply:
Would it be complicated to support this option to keep the block caret in insert mode with this option? In favor of backward compatibility. And this might be a simpler option to use block caret instead of diving into the guicursor option.

Reply:
But if it's complicated, let's leave it without this option.

Reply:
> For 212, this setting is mostly ignored

My mistake. For 211 and 212, this setting is completely ignored :) The rest of the comment is right though - if the user changes the setting, it will be set until something Vim happens and we redraw the caret.

The request originally came from [VIM-1475](https://youtrack.jetbrains.com/issue/VIM-1475). I think 'guicursor' supersedes this, tbh, and so I removed it. It offers more control, and is the Vim way to control Vim behaviour. Personally, I think that if there is an IntelliJ and a Vim way to control Vim behaviour, the Vim way should take precedence. Having two settings provides two sources of truth, and it's not obvious which one wins - if the setting is checked, we'll use a block cursor, but if it's not checked, we'll use whatever 'guicursor' is set to use, which might be block, bar or underscore. If the setting is checked, should it apply to all modes, or just insert?

(An alternative would be some kind of sync mechanism, which would also be useful for things like line numbers and, eventually, wrap. But here it would still be difficult to map between them - would checking the option update 'guicursor' to block in all modes, or just insert, or what? There's also the question of per-`Editor` settings vs global IdeaVim options, but that's a bigger conversation)

Personally, I would prefer not to add the VIM-1475 solution back. That said, it can be done pretty easily, especially if we just say that "use block cursor" is a complete override - no matter what mode, just use block. You're the boss, your call :) Happy to go with your decision.

Reply:
Done. Added support for the "use block caret" setting to override `'guicursor'`. If set, "use block caret" now means "use block caret everywhere", and it is used for normal, insert, replace, and all other modes. If not set, we assume that means "use guicursor" and apply the appropriate caret shape.